packages/graphback-core/src/runtime/CRUDService.ts

```diff
@@ -72,40 +79,51 @@ export class CRUDService<Type = any> implements GraphbackCRUDService<Type>  {
     return result;
   }
 
-  //tslint:disable-next-line: no-reserved-keywords
-  public async delete(data: Type, context: GraphbackContext): Promise<Type> {
-    if (this.crudOptions.subDelete) {
-      context.graphback.options.selectedFields = [];
+  public async delete(data: Type, context?: GraphbackContext, info?: GraphQLResolveInfo): Promise<Type> {
+    let selectedFields: string[];
+    if (info && !this.crudOptions.subDelete) {
+      selectedFields = getSelectedFieldsFromResolverInfo(info, this.model);
     }
 
-    const result = await this.db.delete(data, context);
+    const result = await this.db.delete(data, selectedFields);
 
     if (this.pubSub && this.crudOptions.subDelete) {
-      const topic = this.subscriptionTopicMapping(GraphbackOperationType.DELETE, this.modelName);
+      const topic = this.subscriptionTopicMapping(GraphbackOperationType.DELETE, this.model.graphqlType.name);
       const payload = this.buildEventPayload('deleted', result);
       await this.pubSub.publish(topic, payload);
     }
 
     return result;
   }
 
-  public findOne(args: Partial<Type>, context: GraphbackContext): Promise<Type> {
-    return this.db.findOne(args, context)
-  }
+  public findOne(args: Partial<Type>, context?: GraphbackContext, info?: GraphQLResolveInfo): Promise<Type> {
+    let selectedFields: string[];
+    if (info) {
+      selectedFields = getSelectedFieldsFromResolverInfo(info, this.model);
+    }
 
-  public async findBy(filter: QueryFilter<Type>, context: GraphbackContext, page?: GraphbackPage, orderBy?: GraphbackOrderBy): Promise<ResultList<Type>> {
-    let count: number;
+    return this.db.findOne(args, selectedFields);
+  }
 
-    if (context.graphback.options?.aggregations?.count) {
-      count = await this.db.count(filter);
+  public async findBy(args?: FindByArgs, context?: GraphbackContext, info?: GraphQLResolveInfo, path?: string): Promise<ResultList<Type>> {
+    let selectedFields: string[];
+    let requestedCount: boolean = false;
+    if (info) {
+      selectedFields = getSelectedFieldsFromResolverInfo(info, this.model, path);
+      requestedCount = path === 'items' && getResolverInfoFieldsList(info).some((field: string) => field === "count");
```

Comment:
We should only perform the count aggregation query if path is `items` i.e we are querying for the list of "raw" items with pagination.

e.g if we issue a query like 
```graphql
query {
  getXXX(id: 1) {
   id, 
  yyy {
    id 
    count
  }
 }
}

``

This will perform the count query, while it should not. 

yyy and xxx have a one to one relationship between them.

Reply:
Yep, you are absolutely right. Great eye!

Reply:
I've pushed that fix now:

```ts
 requestedCount = path === 'items' && getResolverInfoFieldsList(info).some((field: string) => field === "count");
```

Reply:
Thanks.